### ISSUE TITLE
Dynamic Method Invocation

### ISSUE THREAD

**Issue by nasser** [nasser/magic#52]
Compile `dynamic-method` nodes.

```clj
(.quz foo a b c)
```

Where the type of `foo` is not known at compile time.

- [x] Implementation
- [x] Initial Tests
- [ ] All Initial Tests Passing

**Comment by nasser** [comment]
fails on 

```clj
(.Substring (identity "hello") 2 3)
```
```
Exception System.Exception: Could not invoke member method `Substring` on target hello with argument types System.Int64, System.Int64.
  at Magic.Dispatch.InvokeMember (System.Object o, System.String name, System.Object[] args) [0x00088] in <af59c918188542cbabbc5f8a67642897>:0 
  at magic$$magic$test$dynamic$$--anonymous--$$8656.invoke () [0x00014] in <11226f68479c4272ac6b6a33259eee32>:0 
  at magic$api$eval__8444.invokeStatic (System.Object ) [0x00034] in <11226f68479c4272ac6b6a33259eee32>:0 
  at magic$api$eval__8444.invoke (System.Object ) [0x00000] in <11226f68479c4272ac6b6a33259eee32>:0 
  at magic$test$dynamic$eval__8648__8653.invokeStatic () [0x0000a] in <11226f68479c4272ac6b6a33259eee32>:0 
  at magic$test$dynamic$eval__8648__8653.invoke () [0x00000] in <11226f68479c4272ac6b6a33259eee32>:0 
  at clojure.lang.Compiler.eval (System.Object form) [0x001c2] in <296f2b0f4fdf405a8d93ba141aa3052f>:0 
  at clojure$core$eval__22334.invokeStatic (System.Object ) [0x00000] in <296f2b0f4fdf405a8d93ba141aa3052f>:0 
  at clojure$core$eval__22334.invoke (System.Object ) [0x00000] in <296f2b0f4fdf405a8d93ba141aa3052f>:0 
  at nostrand$repl$clifn__2508fn__2513__2517.invoke () [0x00053] in <7668a44167a541c4bd10ed464bb5ccb1>:0 
```

The exception is thrown by `Magic.Dispatch.InvokeMember` so its a runtime bug, not a compiler bug.

**Comment by nasser** [comment]
Fixed in 091f8d9a61a39990dee16529bcdce53ea0d91494
